build_tools/benchmarks/common/benchmark_config.py

```diff
@@ -108,12 +108,23 @@ def real_path_or_none(
           capture_tarball=args.capture_tarball.resolve(),
           capture_tmp_dir=per_commit_tmp_dir / CAPTURES_REL_PATH)
 
-    build_dir = args.build_dir.resolve()
-    if args.run_config is None:
-      # TODO(#11076): Remove legacy path.
-      root_benchmark_dir = build_dir / BENCHMARK_SUITE_REL_PATH
+    if args.e2e_test_artifacts_dir is not None:
+      if args.run_config is None:
+        raise ValueError(
+            "--e2e_test_artifacts_dir only supports using with --run_config.")
```

Comment:
Grammar is a bit weird here. Maybe this is what you mean?

```suggestion
            "--e2e_test_artifacts_dir requires --run_config.")
```